### PR TITLE
docs(profiling): add rust profiling alpha doc

### PR DESCRIPTION
**Pull request by viglia**
In preparation for the rust profiling alpha, this adds the rust profiling setup.

### CODE REVIEW

File: src/docs/product/profiling/getting-started.mdx [hunk 1]
```diff
@@ -28,3 +28,4 @@ Profiling depends on Sentry's performance monitoring product being enabled befor
   - [iOS](/platforms/apple/guides/ios/profiling/)
   - [Python](/platforms/python/profiling/)
   - [Node.js](/platforms/node/profiling)
+  - [Rust](/platforms/rust/profiling)
```
Comment (imatwawana) [inline src/docs/product/profiling/getting-started.mdx:31]: Is there any reason to not start putting these in alphabetical order?

File: src/docs/product/profiling/index.mdx [hunk 2]
```diff
@@ -16,6 +16,7 @@ Profiling is currently in beta. Beta features are still in-progress and may have
   * iOS (Swift and Objective-C only)
   * Python
   * Node.js
+  * Rust
```
Comment (imatwawana) [inline src/docs/product/profiling/index.mdx:19]: Same comment as above

File: src/platforms/common/profiling/index.mdx [hunk 3]
```diff
@@ -159,6 +169,35 @@ The <PlatformIdentifier name="profiles_sample_rate" /> setting is *relative* to
 
 </PlatformSection>
 
+
+<PlatformSection supported={["rust"]}>
```
Comment (imatwawana) [inline src/platforms/common/profiling/index.mdx:173]: This section is missing a code snippet for enabling performance for Rust. See PlatformSections for other supported SDKs above.

File: src/platforms/common/profiling/index.mdx [hunk 4]
```diff
@@ -159,6 +169,35 @@ The <PlatformIdentifier name="profiles_sample_rate" /> setting is *relative* to
...
+
+<Note>
+
+Rust profiling alpha is available starting in SDK version `0.29.0`.
```
Comment (imatwawana) [inline src/platforms/common/profiling/index.mdx:177]:
```suggestion
Rust Profiling alpha is available starting in SDK version `0.29.0`.
```

**Review by imatwawana — CHANGES_REQUESTED** [review]
I've made a couple wording tweaks and left some questions. There also seems to be a code snippet missing.

File: src/docs/product/profiling/getting-started.mdx (hunk 1 above)
Comment (viglia) [reply src/docs/product/profiling/getting-started.mdx:31]: no, good point. I'm fixing that

File: src/platforms/common/profiling/index.mdx [hunk 5]
```diff
@@ -159,6 +183,44 @@ The <PlatformIdentifier name="profiles_sample_rate" /> setting is *relative* to
...
+
+<Note>
+
+Rust Profiling alpha is available starting in SDK version `0.29.0`.
+
+</Note>
+
+In order to use Profiling you first need to enable the `profiling` feature in the `sentry` dependency inside the project `Cargo.toml`:
+
+```cargo
```
Comment (viglia) [inline src/platforms/common/profiling/index.mdx:197]: @imatwawana where can I check which syntax (`xml`, `python`, etc.) is supported for these code snippets?

I've put `cargo` here but I'm unaware of whether that is actually supported or not.

ps. thanks for the previous feedback.

Comment (imatwawana) [reply src/platforms/common/profiling/index.mdx:197]: > @imatwawana where can I check which syntax (`xml`, `python`, etc.) is supported for these code snippets?
> 
> I've put `cargo` here but I'm unaware of whether that is actually supported or not.
> 
> ps. thanks for the previous feedback.

I'm not sure where you'd check that for the whole repository. @untitaker do you know? But for Rust, `cargo` is definitely already in use in this repo. Also you can check the preview generated by Vercel to see if it's rendering the way you intend. Here's [that page](https://sentry-docs-git-viglia-rustprofiling.sentry.dev/platforms/rust/profiling/) in the preview so you can confirm. When you're in the preview, you can navigate to any page, but you can't use the search or you'll end up back in the live docs.

**Review by imatwawana — APPROVED** [review]

**Comment by viglia** [comment]
nice, thanks @imatwawana !!

**Pull request merged**